### PR TITLE
Added db_get_param for rendering the .databrickscfg file 

### PR DESCRIPTION
**Pull request by skodman**
- renders the .databrickscfg file for parameters used in brickster

- Modified db_wsid(), db_token() and db_host() to use db_get_param()

### CODE REVIEW

**Comment by zacdav-db** [comment]
I think the flow should work slightly differently to what is currently proposed.

Want to support the reading of both a `.databrickscfg` file and also `.Renviron` which is the default and surface the switching of those ideally through setting of options.

My current thinking is similar to this:

```r
db_host <- function(id = NULL, prefix = NULL, env = getOption("db_env")) {
  
  # if option `use_databrickscfg` is `TRUE` then fetch the associated env.
  # env is specified via `db_env` option, if missing use default.
  # this behaviour can only be changed via setting of config
  if (getOption("use_databrickscfg", FALSE)) {
  
    # TODO:
    # depending on value of `env` get appropriate data from `.databrickscfg`
     
  }

  if (is.null(id) && is.null(prefix)) {
    
    if (is.null(env)) {
      host <- Sys.getenv("DATABRICKS_HOST")
    } else {
      host <- Sys.getenv(paste("DATABRICKS_HOST", upper(env), sep = "_"))
    }
    
    if (host == "") {
      stop(format_error(c(
        "`DATABRICKS_HOST` not found in `.Renviron`:",
        "x" = "Need to specify `DATABRICKS_HOST` within `.Renviron` file."
      )))
    }
  } else {
    host <- paste0("https://", prefix, id, ".cloud.databricks.com")
  }
  
  host
}
```
In this case `db_host()` would be expected to behave is does today without change unless specifying options.

Setting `option(db_env = "prod")` would then search for `DATABRICKS_HOST_PROD` in `.Renviron`.

Setting both `option(db_env = "prod")` and `option(db_env = "use_databrickscfg")` would search `.databrickscfg` for host under the "prod" profile.

Setting only `option(db_env = "use_databrickscfg")` would use the default profile's host in the `.databrickscfg` file.

File: R/package-auth.R [hunk 1]
```diff
@@ -72,16 +73,17 @@ db_token <- function() {
 #' @import cli
 #' @export
 db_wsid <- function() {
-  token <- Sys.getenv("DATABRICKS_WSID")
+  #token <- Sys.getenv("DATABRICKS_WSID")
+  wsid=db_get_param('dbWsId')
 
-  if (token == "") {
+  if (wsid == "") {
```
Comment (zacdav-db) [inline R/package-auth.R:79]: Good catch.

File: R/package-auth.R [hunk 2]
```diff
@@ -18,7 +18,8 @@
 #' @export
 db_host <- function(id = NULL, prefix = NULL) {
   if (is.null(id) && is.null(prefix)) {
-    host <- Sys.getenv("DATABRICKS_HOST")
+    #host <- Sys.getenv("DATABRICKS_HOST")
+    host=db_get_param('dbHost')
```
Comment (zacdav-db) [inline R/package-auth.R:22]: ideally the fetching of param would only be offloaded in circumstance of using `.databrickscfg`, at least that's my current thinking.

File: R/package-auth.R [hunk 3]
```diff
@@ -46,8 +47,8 @@ db_host <- function(id = NULL, prefix = NULL) {
 #' @import cli
 #' @export
 db_token <- function() {
-  token <- Sys.getenv("DATABRICKS_TOKEN")
-
+  #token <- Sys.getenv("DATABRICKS_TOKEN")
```
Comment (zacdav-db) [inline R/package-auth.R:50]: Same expectation as commented around `db_host()`.

File: R/package-auth.R [hunk 4]
```diff
@@ -72,16 +73,17 @@ db_token <- function() {
 #' @import cli
 #' @export
 db_wsid <- function() {
-  token <- Sys.getenv("DATABRICKS_WSID")
+  #token <- Sys.getenv("DATABRICKS_WSID")
+  wsid=db_get_param('dbWsId')
```
Comment (zacdav-db) [inline R/package-auth.R:77]: Same expectation as commented around db_host().

**Comment by skodman** [comment]
OK, let me summarise what i understand from your suggestions:

1. Both **.Renviron** and **.databrickscfg** should be supported
2. Switching between different configs is controlled through two options:**db_env** and **use_databrickscfg**.
3. If option **use_dbconfig** is not set (it's NULL), the we use **.Renviron**
4. The option **db_env** is used to either switch between profiles in **.databrickscfg** or to select variable from **.Renviron** defined for that specific environment (by adding a suffix **_<db_env>** i.e. DATABRICKS_HOST **_PROD**

The side effect of this approach is to have exactly the same profile names and environment suffixes.

File: R/package-auth.R (hunk 3 above)
Comment (skodman) [reply R/package-auth.R:50]: added env argument defaulting to the option'db_env'. If NULL db_get_parm will use DEFAULT in both cases (.databrickscfg and .Renviron) which is explaiend in below comment

**Comment by zacdav-db** [comment]
Thanks @skodman.

I've been mulling over the changes and I've decided on a direction, I'm going to merge these changes into the `multi-env-support` branch and then tweak things before re-raising a PR into main.

Once I've made my changes will still need help to add unit tests and check behaviour on windows before this reaches main branch.

**Comment by zacdav-db** [comment]
Refer to https://github.com/zacdav-db/brickster/pull/22 for new pull request.

**Pull request merged**